### PR TITLE
[Order List Redesign] Show Processing and All Orders Tabs

### PR DESCRIPTION
**Pull request by shiki**
Refs #956. 

This implements the new tabs, "Processing" and "All Orders", within the Orders tab. This was made easier after the refactoring that was completed in #1816 and #1835. 

The search UI that was completed in #1805 is now **activated for all users**. The `.orderListRedesign` feature flag was removed.  

## Video

[Here](https://drive.google.com/file/d/130-tauhEfFGYxm0UCIZSqxAQqSsfhP_H/view?usp=sharing) is a video of how the Order List works so far. 

## Screenshots

Light | Dark
--------|-------
image        |       image

Large Font ❗️ | Landscape 
--------|-------
image        |       image

### Accessibility

On large font sizes, the tab isn't quite visible. I'll tackle this separately. It looks like the problem is similar to the _My Store_ as well. 

### Empty States

The empty states will be tackled in a separate PR. 

### Landscape Issue

There's an existing issue with the `XLPagerTabStrip` on landscape mode. Sometimes the view gets clipped. I spent more than an hour trying to figure it out. Solutions that I found that fixes this:

- Make `OrdersViewController` is a subclass of `UITableViewController`.
- Make `OrdersViewController.xib` use `UITableView` as its root.

Both solutions break the empty states. I'll look more into this next. But with this issue and the accessibility issue described above, I'm sort of convinced that we should probably implement our own component so that we have more control over these. 🙁 

## Design Review 🎨 

@Garance91540 Please take a look at the video and the screenshots above. How does it look so far? Aside from the empty states, is there something that should be fixed?

## Testing

1. Log in and navigate to Orders.
2. Tap on one of the tabs. Are they filtered correctly?
3. Switch to a different store. Did the list refresh? 
4. Reinstall the app. Did the Order Statuses and Orders load correctly?
5. Tap on Orders → Search. Tap on a filter. Are the displayed Orders filtered by what you picked?

## Reviewing 

Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist

- [x] If it's feasible, I have added unit tests. 
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by pmusolino** [comment]
Thanks for working on this @shiki!
I see all the problem around `XLPagerTabStrip`, and I think that the best solution here is to implement our custom component, that maybe can be used also for the Dashboard UI (if flexible enough).
More context here p99K0U-1Cj

What do you think @jaclync? Do you think that it's possible to easily substitute `XLPagerTabStrip` with our custom component, which maybe can be implemented first here, and after in the Dashboard UI?

**Comment by shiki** [comment]
@Garance91540 Here are some screenshots for the iPad. Let me know if you need more!

_ | _
--------|-------
image        |       image
image | -  

> On each Order row, can we make sure the date's format is Month Day, Year? Right now I'm just seeing Month Day

I actually followed Android on this. The conditions are:

- If it's in the current year, use Month and Day only
- If not, use Month, Day, and Year.

File: WooCommerce/Classes/ViewRelated/Orders/OrdersMasterViewController.swift [hunk 1]
```diff
@@ -107,6 +74,36 @@ final class OrdersMasterViewController: UIViewController {
...
+    /// Return the ViewControllers for "Processing" and "All Orders".
+    ///
+    override func viewControllers(for pagerTabStripController: PagerTabStripViewController) -> [UIViewController] {
+        // TODO This is fake. It's probably better to just pass the `slug` to `OrdersViewController`.
+        let processingOrderStatus = OrderStatus(
+            name: OrderStatusEnum.processing.rawValue,
+            siteID: Int64.max,
+            slug: OrderStatusEnum.processing.rawValue,
+            total: 0
+        )
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/OrdersMasterViewController.swift]: do we plan on addressing this TODO in a followup task? feel free to refactor `OrderStatus`'s init as a separate task

**Comment by jkmassel** [comment]
I'm freezing `3.7`, so this is being bumped to `3.8`. If you need it to be part of the `3.7` release, please re-target `release/3.7` and merge there. Once you have, please ping me that it's been merged and I'll be happy to cut a new beta release!

File: WooCommerce/Classes/ViewRelated/Orders/OrdersMasterViewController.swift (hunk 1 above)
Comment (shiki) [reply WooCommerce/Classes/ViewRelated/Orders/OrdersMasterViewController.swift]: Yes. It's on my list. I also added it to https://github.com/woocommerce/woocommerce-ios/issues/956.

File: WooCommerce/Classes/System/DefaultFeatureFlagService.swift [hunk 2]
```diff
@@ -13,8 +13,6 @@ struct DefaultFeatureFlagService: FeatureFlagService {
             return true
         case .refunds:
             return BuildConfiguration.current == .localDeveloper
-        case .orderListRedesign:
-            return BuildConfiguration.current == .localDeveloper
```
[the remark this answers is not in the document]
Comment (shiki) [reply WooCommerce/Classes/System/DefaultFeatureFlagService.swift]: I'm not sure. This change removes the option to go back to the old Orders List. I guess it should be deleted?

[the remark this answers is not in the document]
Comment (jaclync) [reply WooCommerce/Classes/System/DefaultFeatureFlagService.swift]: personally I always like having the option to hide a feature if needed 🙂 but if all the tasks are done for the feature and it's ready for release, go for it!

**Comment by shiki** [comment]
> I was wondering if it's the following code that was missing when you tried the container?

Thank you, @jaclync. That was it. I didn't think about doing that. 🙏 I cherry-picked your commit to this branch.

---

Going to merge this. Most of the design issues were fixed in #1903 and #1861. This is what's left:

> Can we add a border under the "Processing/All Orders" Tab? It avoid the tabs to blur with the order rows on scrolling.

I'll work on that on a separate PR.

**Pull request merged**